jerry-core/parser/js/js-scanner-util.c

```diff
@@ -1781,6 +1781,42 @@ scanner_is_global_context_needed (parser_context_t *context_p) /**< context */
   return false;
 } /* scanner_is_global_context_needed */
 
+/**
+ * Try to scan/parse the ".target" part in the "new.target" expression.
+ *
+ * Upon exiting with "true" the current token will point to the "target"
+ * literal.
+ *
+ * If the method returns with "false" the context and current token is
+ * restored to the "new" keyword.
+ *
+ * @returns true if the ".target" part was found, otherwise false.
+ */
+bool
+scanner_try_scan_new_target (parser_context_t *context_p) /**< parser/scanner context */
+{
+  lexer_token_t original_token = context_p->token;
+  JERRY_ASSERT (original_token.type == LEXER_KEYW_NEW);
+
+  scanner_location_t location;
+  scanner_get_location (&location, context_p);
+  lexer_next_token (context_p);
```

Comment:
This is too complex. There is a function to test the next character.

Reply:
I need to test the next character and next literal. Is there a method for it?

Reply:
How about this?

```
if (lexer_check_next_character (context_p, LIT_CHAR_DOT))
      {
        context_p->source_p++;
        context_p->token.flags &= (uint8_t) ~LEXER_NO_SKIP_SPACES;
        lexer_skip_spaces (context_p);
        /* 'new.target' must not contain escaped characters */
        if (context_p->source_end_p - context_p->source_p >= 6
            && memcmp (context_p->source_p, "target", 6) == 0)
        {
          // emit new.target opcode
        }
        /* new.<anything> is an invalid primary expression> */
        else
        {
          parser_raise_error (context_p, PARSER_ERR_PRIMARY_EXP_EXPECTED);
        }
      }
```

Reply:
Use consumeCharacter because the column needs to be increased as well. As far as I know, there cannot be anything else after a `new.` except `target`, so you can use `lexer_next_token` after the dot and throw an error on fail. The prposed solution is also wrong because `new.targetX` is also accepted as well.